### ISSUE TITLE
bug: `✗ type preservation failed at test/Tests/EvalFull.hs:816:23`

### ISSUE THREAD

**Issue by dhess** [hackworthltd/primer#1036]
### Code of conduct

- [X] I've read the project's code of conduct, and agree to follow it.

### Don't report security issues here

- [X] I've read the project's [security policy](https://github.com/hackworthltd/primer/security/policy), and I do not believe this bug is a security issue.

### Is there an existing issue?

- [X] I've searched GitHub Issues, and I believe this is a new issue

### What happened?

Test failure, partial log is:

```
      type preservation:                                  FAIL (9.51s)
          ✗ type preservation failed at test/Tests/EvalFull.hs:816:23
            after 884 tests, 16 shrinks and 546 discards.
            long TimedOut               1% ▏···················
            mid TimedOut               20% ███▉················
            long NF                    99% ███████████████████▋
            mid skipped due to LetType  1% ▏···················
            generated a normal form    80% ████████████████····
          
               ┏━━ test/Tests/Eval/Utils.hs ━━━
            49 ┃ genDirTm :: PropertyT WT (Dir, Expr, Type' ())
            50 ┃ genDirTm = do
...
            828 ┃           midSteps <- forAllT $ Gen.integral $ Range.linear 1 (steps - 1)
            829 ┃           (_, s') <- failWhenSevereLogs $ evalFullStepCount @EvalLog tds globs midSteps dir t
            830 ┃           test "mid " s'
          
            This failure can be reproduced by running:
            > recheck (Size 29) (Seed 5955399570369228765 9348635574744870875) type preservation
          
        Use '--pattern "$NF ~ /type preservation/" --hedgehog-replay "Size 29 Seed 5955399570369228765 9348635574744870875"' to reproduce from the command-line.
        
        Use -p '$0=="test/Test.hs.Tests.EvalFull.type preservation"' to rerun this test only.
```

Full log here:

* https://gist.github.com/dhess/23f1db335a73a317cf615bf05b26a3da

### What should have happened?

Tests should have passed.

### Which package(s) or aspect(s) of the project are the source of the issue?

primer

### Any additional information?

This is on a development branch, but the `main` it's based on is 51b3abc2eaa03afcaacc6ea07299273f8dac92de

**Comment by brprice** [comment]
Due to the fact that this in based on https://github.com/hackworthltd/primer/commit/51b3abc2eaa03afcaacc6ea07299273f8dac92de which does not include #1028
and the following snippet of log
```
                ┃                 │ CaseOfHoleNeedsEmptyBranches
                ┃                 ^^^^^^^^^^^^^^^^^^^^
```
I am fairly confident that this is the same issue as #1034 and has already been fixed by #1028. Sorry I didn't catch this sooner!

**Comment by dhess** [comment]
No worries, of course. Shall I close it, then?
